### PR TITLE
chore: upgrade kotlin to 1.7.20

### PR DESCRIPTION
**Pull request by vitorhugods**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

We're stuck in the past with Kotlin 1.6.10 and old versions of libraries such as Ktor, Turbine, Okio, and so many others have jumped ahead of us with plenty of bugfixes that we're having to work-around.

### Causes

We have hold back for a while because of https://youtrack.jetbrains.com/issue/KT-52172 which is a bit annoying and makes the IDE go bananas, but at this point, the costs of holding are outweighing the costs of going forward.

### Solutions

Upgrade. Kotlin from `1.6.10` to `1.7.20` and all dependencies we needed to make it work as well.

Some breaking changes identified are that:
- ✅ `when` branches must be exhaustive
- ✅ SQLDelight changed the package name
- ✅ SQLDelight requires to pass a context/dispatcher when mapping flows (I added a default that mimics the old behaviour, but we should refactor to inject a dispatcher from the Database instead and use a Test Dispatcher for tests)
- ✅ Ktor had a minor package name change for `ContentNegotiation`
- ⚠️ (we can do later if needed) PBandK/Protobuf is broken for JavaScript, we need to investigate further. 

Some work-arounds for issues we added and can be removed:
- ✅ wrapping `Turbine` around a UnconfinedTestDispatcher is not necessary anymore
- ⌛ (we can do later) Remove refreshing auth token before connecting to the websocket (websocket should do it by itself)
- ⌛ (we can do later) Remove our own implementation of SqlDelight's Paging library and use the newer version which should work out of the box
- ⌛ (we can do later) - Upgrade the LibSodium wrapper to a newer version

> **Note**: After this merge, Reloaded developers won't be able to just edit Kalium files from the same IDE window and will need to open a dedicated IDE to change Kalium files, as type resolution will be broken.

### Testing

N/A ?

Compiler should compile.

I'm running Reloaded using Kalium on this branch and so far so good.

----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
2. https://github.com/wireapp/.github#usage
3. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/sync/SlowSyncRepositoryTest.kt [hunk 1]
```diff
@@ -43,7 +42,7 @@ class SlowSyncRepositoryTest {
     }
 
     // TODO: Re-enable once we can update Turbine to 0.11.0+ (Requires Kotlin 1.6.21+)
-    @Ignore
+//     @Ignore
```
Comment (MohamadJaara) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/data/sync/SlowSyncRepositoryTest.kt:45]: we can remove the Ignore 🚀

**Comment by yamilmedina** [comment]
Question, Should we merge it after code freeze, right ?

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/data/sync/SlowSyncRepositoryTest.kt (hunk 1 above)
Comment (vitorhugods) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/data/sync/SlowSyncRepositoryTest.kt:45]: Oooh yeah. I was commenting out just to test it and forgot to remove 😄

**Comment by vitorhugods** [comment]
Adding the `DO NOT MERGE` as we need to address the fact that `Verify DB Migrations` is taking more than 1h now.

**Pull request merged**